generator/spring-boot/src/main/java/io/fabric8/maven/generator/springboot/SpringBootGenerator.java

```diff
@@ -52,6 +60,40 @@ public boolean isApplicable(List<ImageConfiguration> configs) {
     }
 
     @Override
+    public List<ImageConfiguration> customize(List<ImageConfiguration> configs) throws MojoExecutionException {
+        // if enabled lets generate a spring devtools token
+        String flag = getProjectProperty("fabric8.spring-devtools.generateToken", null);
+        boolean generateToken = !Strings.isNullOrEmpty(flag) && Configs.asBoolean(flag);
+        if (generateToken || getContext().runningWithGoal("fabric8:watch-spring-boot", "fabric8:watch")) {
+            generateSpringDevToolsToken();
+        }
+        return super.customize(configs);
+    }
+
+    private void generateSpringDevToolsToken() throws MojoExecutionException {
+        Properties properties = MavenUtil.getSpringBootApplicationProperties(getProject());
+        String remoteSecret = properties.getProperty(DEV_TOOLS_REMOTE_SECRET);
+        if (Strings.isNullOrEmpty(remoteSecret)) {
+            String newToken = UUID.randomUUID().toString();
+            log.info("Generating the spring devtools token: " + newToken);
```

Comment:
Since this info is normally not of much value for the daily usage (correct me when I'm wrong)  maybe its better to move this to level 'verbose', i.e. `log.verbose()` which can be enabled with `-Dfabric8.verbose`. Less noisy than debug, and helps in finding errors.